core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -174,6 +245,194 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
       case r @ Right(_) => F.pure(EitherUtil.leftCast(r))
     })
 
+  /**
+   * Handles errors by materializing them into `Either` values.
+   *
+   * This is the implementation of [[ApplicativeError.attempt]].
+   *
+   * Example:
+   * {{{
+   *   parseNum(x).attempt.map {
+   *     case Right(num) => num
+   *     case Left(_) => 0L
+   *   }
+   * }}}
+   *
+   * @see [[attemptF]] for recovering errors thrown in `F[_]`
+   *
+   * @see [[redeem]] and [[redeemWith]] for optimizations on `attempt.map`
+   *      and `attempt.flatMap` respectively
+   */
+  def attempt(implicit F: Functor[F]): EitherT[F, A, Either[A, B]] =
+    EitherT.right(value)
+
+  /**
+   * Handles errors thrown in `F[_]` (if it implements [[ApplicativeError]]),
+   * by materializing them into `Either` values.
+   *
+   * $monadErrorF
+   *
+   * Example:
+   * {{{
+   *   import cats.effect.IO
+   *
+   *   val rio = EitherT.pure[IO, String](10)
+   *
+   *   // Yields Right(10)
+   *   rio.attemptF
+   *
+   *   val dummy = new RuntimeException("dummy")
+   *   val lio = EitherT[IO, String, Int](IO.raiseError(dummy))
+   *
+   *   // Yields Left(RuntimeException("dummy"))
+   *   lio.attemptF
+   * }}}
+   *
+   * Note that in this sample we are materializing the `Throwable` of the
+   * underlying `IO`, even the source `EitherT` value is not parametrized
+   * with it.
+   *
+   * @see [[attempt]] for recovering errors expressed via `EitherT`
+   *
+   * @see [[redeemF]] and [[redeemWithF]] for optimizations on `attemptF.map`
+   *      and `attemptF.flatMap` respectively
+   */
+  def attemptF[E](implicit F: ApplicativeError[F, E]): EitherT[F, A, Either[E, B]] =
+    EitherT(F.map(F.attempt(value)) {
+      case Right(Right(b)) => Right(Right(b))
+      case Right(Left(a))  => Left(a)
+      case Left(e)         => Right(Left(e))
+    })
+
+  /**
+   * Handle any error, potentially recovering from it, by mapping it via
+   * the provided function.
+   *
+   * This is the implementation of [[ApplicativeError.handleError]].
+   *
+   * Example:
+   * {{{
+   *   parseNum(s).handleError(_ => 0L)
+   * }}}
+   *
+   * @see [[handleErrorWith]] for recovering errors by mapping them to
+   *      `EitherT` values (aka the equivalent of `flatMap` for errors)
+   *
+   * @see [[handleErrorF]] for recovering errors thrown in `F[_]`
+   */
+  def handleError(f: A => B)(implicit F: Functor[F]): EitherT[F, A, B] =
```

Comment:
I kind of think this should take a new type parameter for A:

```
def handleError[A1](f: A => B)... : EitherT[F, A1, B]
```
Since we know that there is no A left in the result.